addon/test/test-metrics.js

```diff
@@ -0,0 +1,348 @@
+/*
+ * This Source Code is subject to the terms of the Mozilla Public License
+ * version 2.0 (the "License"). You can obtain a copy of the License at
+ * http://mozilla.org/MPL/2.0/.
+ */
+const self = require('sdk/self');
+const { before } = require('sdk/test/utils');
+const MockUtils = require('./lib/mock-utils');
+
+const BROWSER_DURATION = 60 * 60;
+const EVENT_SEND_METRIC = 'testpilot::send-metric';
+const EVENT_RECEIVE_VARIANT_DEFS = 'testpilot::register-variants';
+const EVENT_SEND_VARIANTS = 'testpilot::receive-variants';
+const TELEMETRY_TESTPILOT = 'testpilot';
+const TELEMETRY_EXPERIMENT = 'testpilottest';
+const PREFERENCE_OVERRIDES = {
+  'toolkit.telemetry.enabled': true,
+  'datareporting.healthreport.uploadEnabled': true
+};
+
+// TODO: Remove this / switch based on --verbose switch?
+MockUtils.setDebug(true);
+
+const mocks = {
+  store: {},
+  callbacks: MockUtils.callbacks({
+    seedrandom: ['seedrandom'],
+    Events: ['on', 'off', 'emit'],
+    TelemetryController: ['submitExternalPing'],
+    AddonManager: ['getAddonByID'],
+    PrefsService: ['set', 'get']
+  })
+};
+
+const mockLoader = MockUtils.loader(module, './lib/metrics.js', {
```

Comment:
I couldn't get something like that to work, since this is using the SDK's require() implementation and I can't really tweak that much.